unicorn/include/cocaine/idl/unicorn.hpp

```diff
@@ -0,0 +1,267 @@
+/*
+* 2015+ Copyright (c) Anton Matveenko <antmat@yandex-team.ru>
+* All rights reserved.
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*/
+
+#ifndef COCAINE_UNICORN_SERVICE_INTERFACE_HPP
+#define COCAINE_UNICORN_SERVICE_INTERFACE_HPP
+
+#include "cocaine/unicorn/path.hpp"
+#include "cocaine/unicorn/value.hpp"
+
+#include <cocaine/rpc/protocol.hpp>
+#include <boost/mpl/list.hpp>
+#include <vector>
+
+namespace cocaine { namespace io {
+
+struct unicorn_tag;
+struct unicorn_final_tag;
+struct unicorn_locked_tag;
+
+using namespace cocaine::unicorn;
```

Comment:
Using namespaces in header file (even if enclosed in other namespace) smells bad, isn't it?